auth/router.js

```diff
@@ -0,0 +1,29 @@
+const express = require('express');
+const passport = require('passport');
+const jwt = require('jsonwebtoken');
+
+const config = require('../config');
+
+const createAuthToken = user => {
+  return jwt.sign({user}, config.JWT_SECRET, {
+    subject: user.userName,
+    expiresIn: config.JWT_EXPIRY,
+    algorithm: 'HS256'
+  });
+};
+
+const router = express.Router();
+
+// The user provides a username and password to login
+router.post('/login', passport.authenticate('basic', {session: false}), (req, res) => {
```

Comment:
rather than using basic here, pass username/password as parameters to the post, that should simplify the issues you are seeing.